python_ta/reporters/plain_reporter.py

```diff
@@ -81,3 +82,33 @@ def _colour_messages_by_type(self, messages: Dict[str, List[NewMessage]]) -> str
                 result += self._BREAK
 
         return result
+
+    def _add_line(self, lineno: int, linetype: LineType, slice_: slice, text: str = "") -> str:
+        """Format given source code line as specified and return as str.
+
+        Called by _build_snippet, relies on _colourify.
+        """
+        snippet_so_far = super()._add_line(lineno, linetype, slice_, text)
+        if linetype == LineType.ERROR:
+            start = slice_.start or 0
+            prespace = (
+                7 + start
+            ) * self._SPACE  # number 7 for prespaces included for adding line number
+            snippet_so_far = self._overline_helper(snippet_so_far, text[slice_], prespace)
+
+        elif linetype == LineType.DOCSTRING:
+            prespace = (7 + len(text) - len(text.lstrip(" "))) * self._SPACE
+            snippet_so_far = self._overline_helper(snippet_so_far, text.lstrip(" "), prespace)
+
+        return snippet_so_far
+
+    def _overline_helper(self, snippet: str, text: str, prespace: str) -> str:
```

Comment:
Now that you've simplified this function a little, the `snippet` parameter really isn't necessary. You can just return the _new line_ and then add it to `snippet_so_far` in `_add_line`